idaes/models_extra/power_generation/unit_models/cross_flow_heat_exchanger_1D.py

```diff
@@ -0,0 +1,815 @@
+                enth_out = b.hot_side.properties[t, z1].get_enthalpy_flow_terms(p_hot)
+
+                return pyunits.convert(
+                    enth_in - enth_out,  # Hot side loses enthalpy
+                    to_units=shell_units["power"],  # Hot side isn't always the shell
+                )
+
+            @self.Expression(self.flowsheet().config.time)
+            def log_mean_delta_temperature(b, t):
+                dT0 = b.hot_side.properties[t, z0].temperature - pyunits.convert(
+                    b.cold_side.properties[t, z0].temperature,
+                    to_units=shell_units["temperature"],
+                )
+                dT1 = b.hot_side.properties[t, z1].temperature - pyunits.convert(
+                    b.cold_side.properties[t, z1].temperature,
+                    to_units=shell_units["temperature"],
+                )
+                return (dT0 - dT1) / log(dT0 / dT1)
+
+            @self.Expression(self.flowsheet().config.time)
+            def overall_heat_transfer_coefficient(b, t):
+                return b.total_heat_duty[t] / (
+                    b.total_heat_transfer_area * b.log_mean_delta_temperature[t]
+                )
+
+    def initialize_build(
+        blk,
+        shell_state_args=None,
+        tube_state_args=None,
+        outlvl=idaeslog.NOTSET,
+        solver="ipopt",
+        optarg=None,
+    ):
+        """
+        CrossFlowHeatExchanger1D initialization routine
+
+        Keyword Arguments:
+            state_args : a dict of arguments to be passed to the property
+                         package(s) to provide an initial state for
+                         initialization (see documentation of the specific
+                         property package) (default = None).
+            outlvl : sets output level of initialization routine
+
+                     * 0 = no output (default)
+                     * 1 = return solver state for each step in routine
+                     * 2 = return solver state for each step in subroutines
+                     * 3 = include solver output information (tee=True)
+
+            optarg : solver options dictionary object (default={'tol': 1e-6})
+            solver : str indicating which solver to use during
+                     initialization (default = 'ipopt')
+
+        Returns:
+            None
+        """
+        init_log = idaeslog.getInitLogger(blk.name, outlvl, tag="unit")
+        solve_log = idaeslog.getSolveLogger(blk.name, outlvl, tag="unit")
+
+        if optarg is None:
+            optarg = {}
+        opt = get_solver(solver, optarg)
+
+        hot_side = blk.hot_side
+        cold_side = blk.cold_side
+        t0 = blk.flowsheet().time.first()
+        if not (
+            "temperature" in hot_side.properties[t0, 0].define_state_vars().keys()
+            and "temperature" in cold_side.properties[t0, 0].define_state_vars().keys()
+        ):
+            raise NotImplementedError(
```

Comment:
@andrewlee94 do you have suggestions on the error message below, or is this fine for a user-facing message?

Reply:
I think it is fine, especially for a contrib model.